src/main/java/com/hazelcast/remotecontroller/HazelcastCloudManager.java

```diff
@@ -0,0 +1,355 @@
+        if(notSetEnvironmentVariables.isEmpty())
+            loginToHazelcastCloud(System.getenv("BASE_URL"), System.getenv("API_KEY"), System.getenv("API_SECRET"));
+        else
+            throw new CloudException("Not all required environment variables are set. These are not set ones: " + notSetEnvironmentVariables);
+    }
+
+    public void loginToHazelcastCloud(String url, String apiKey, String apiSecret) throws CloudException {
+        uri = URI.create(url + "/api/v1");
+        baseUrl = url;
+        bearerToken = getBearerToken(apiKey, apiSecret);
+        if(bearerToken == null)
+            throw new CloudException("Login failed");
+    }
+
+    public CloudCluster createHazelcastCloudStandardCluster(String hazelcastVersion, boolean isTlsEnabled) throws CloudException {
+        try {
+            String clusterName = "test-cluster-" + UUID.randomUUID();
+            String query = String.format("{\"query\":\"mutation {createStarterCluster(input: {name: \\\"%s\\\" cloudProvider: \\\"%s\\\" region: \\\"%s\\\" clusterType: SMALL totalMemory: 2 hazelcastVersion: \\\"%s\\\" isTlsEnabled: %b } ) { id name hazelcastVersion isTlsEnabled state discoveryTokens {source,token} } }\"}",
+                    clusterName,
+                    "aws",
+                    "us-west-2",
+                    hazelcastVersion,
+                    isTlsEnabled);
+
+            LOG.info(String.format("Request query: %s", query));
+            String responseBody = prepareAndSendRequest(query).body().string();
+            LOG.info(maskValueOfToken(responseBody));
+            JsonNode rootNode = mapper.readTree(responseBody).get("data").get("createStarterCluster");
+            String clusterId = rootNode.get("id").asText();
+            if(!waitForStateOfCluster(clusterId, "RUNNING", TimeUnit.MINUTES.toMillis(timeoutForClusterStateWait)))
+                throw new Exception("Wait for cluster state is not finished as expected");
+            return getHazelcastCloudCluster(clusterId);
+        } catch (Exception e) {
+            throw new CloudException("Create cluster failed, error message: " + e.getMessage());
+        }
+    }
+
+    public boolean scaleUpDownHazelcastCloudStandardCluster(String clusterId, int incrementalNodesCount) throws CloudException {
+        int currentMemory;
+        String requestUrl = String.format("%s/cluster/%s/updateMemory", baseUrl, clusterId);
+        try {
+            String query = String.format("{ \"query\": \"query { cluster(clusterId: \\\"%s\\\") { specs {totalMemory} } }\" }", clusterId);
+            String responseBody = prepareAndSendRequest(query).body().string();
+            currentMemory = mapper.readTree(responseBody).get("data").get("cluster").get("specs").get("totalMemory").asInt();
+
+            if (currentMemory + incrementalNodesCount <= 0) {
+                LOG.warn("Scaling number is not proper, there is no enough memory and nodes to scale down");
```

Comment:
Please add all the details you can to the message. E.g. clusterId, incrementalNodesCount, currentMemory, etc.

Reply:
Removed that part.